rust/noosphere-fs/src/fs.rs

```diff
@@ -277,7 +277,12 @@ where
     ) -> Result<Cid> {
         let sphere = Sphere::at(&self.sphere_revision, &self.db);
 
-        // TODO(#85): Error result if mutation is empty and also additional headers are empty
+        let mutation_to_save = self.mutation.get().is_some() && !self.mutation.get().expect("").is_empty();
+        let headers_to_save = additional_headers.is_some() && additional_headers.as_ref().expect("").len() > 0;
```

Comment:
As a rule of thumb, we avoid using `expect` outside of tests (although there are perfectly valid non-test usages for it). On an `Err` result or `None` option, `expect(...)` will panic. But, panic is meant to indicate an unrecoverable error and is not appropriate if we think the caller may be able to recover. Instead, consider using a conditional to progressively unwrap the value and then throw the error later:

```rs
if let Some(mut headers) = additional_headers {
  revision.memo.headers.append(&mut headers);
} else if (mutation.is_empty()) {
  return Err(anyhow!("No changes to save"))
}
```

Reply:
Thanks for the hints. I have updated this. Your suggestion does not check if the vector is there but empty. I have added a guard for this in my updated commit. If this is undesired behavior, I will remove this condition and the associated test again.

Reply:
Yah, thanks for the callout there. It's a weird case where the vector is empty but the value is `Some(...)`, but good to handle it.